tapgarden/planter.go

```diff
@@ -624,6 +625,18 @@ func (c *ChainPlanter) gardener() {
 
 				case err := <-caretaker.cfg.BroadcastErrChan:
 					req.Error(err)
+					// Unrecoverable error, stop caretaker
+					// directly. The pending batch will not
+					// be saved.
+					stopErr := caretaker.Stop()
+					if stopErr != nil {
+						log.Warnf("unable to stop "+
+							"caretaker "+
+							"gracefully: %v", err)
+					}
+
+					delete(c.caretakers, batchKeySerial)
+					c.pendingBatch = nil
```

Comment:
Is the required UX here that the user will repeat all the minting calls in order for the batch to be formed again?

Could also feed the seedlings of that batch back into `c.seedlingReqs` and let the process repeat healthily?

Reply:
I think so? I'm not sure if always trying to re-submit and finalize the batch would help in all cases, or how that won't lead to an infinite loop in cases of an error `tapd` can't resolve like insufficient funds in the backing `lnd` wallet.

Maybe it's worth trying to write this failed batch to disk, and we could later add a way to easily resubmit batches we have stored that are in a cancelled state.


So resubmission would still be manual but at least you wouldn't have to recreate each mint request.

Reply:
Actually this batch will already be on disk, just not with a `Cancelled` state. This feels like a part of #710 tbh. Going to leave as-is since the `Cancelled` states signal intentional failure to finalize a batch vs. an error that caused finalization to fail.